flaskbb/templates/macros.html

```diff
@@ -308,12 +308,52 @@
     {%- endif -%}
 {% endmacro %}
 
-{% macro navlink(endpoint, name, icon='', active='') %}
+{% macro navlink(endpoint, name, icon='', active='', urlforkwargs=None) %}
 <li {% if endpoint == request.endpoint or endpoint == active or active == True %}class="active"{% endif %}>
-    <a href="{{ url_for(endpoint) }}">{% if icon %}<i class="{{ icon }}"></i> {% endif %} {{ name }}</a>
+    <a href="{% if urlforkwargs %}{{ url_for(endpoint, **urlforkwargs) }}{% else %}{{ url_for(endpoint) }}{% endif %}">
+        {% if icon %}<i class="{{ icon }}"></i> {% endif %} {{ name }}
+    </a>
+</li>
+{% endmacro %}
+
+{% macro externalnavlink(uri, name, icon='') %}
+<li>
+    <a href="{{uri}}">{% if icon %}<i class="{{ icon }}"></i> {% endif %} {{ name }}</a>
 </li>
 {% endmacro %}
 
+{% macro navtext(text, icon="", cls="") %}
+<li{% if cls %} class="{{ cls }}"{% endif %}><a href="#">{% if icon %}<i class="{{ icon }}"></i> {% endif %}{{ text }}</a></li>
+{% endmacro %}
+
+{% macro navheader(text, icon="", cls="nav-header") %}
+{{ navtext(text, icon, cls) }}
+{% endmacro %}
+
+{% macro navdivider() %}
+<li class="nav-divider"></li>
+{% endmacro %}
+
+{% macro sidebar(items, extra_class="") %}
```

Comment:
Yeah a generalized navbar macro would be nice but I have this feeling that we couldn't use a generalized macro on our other navigations.

Reply:
Yeah, was playing with the dropdown menu using this rather than being hardcoded and it proved a little difficult. :/ I'll poke at it a bit more and see what I come up with.